addons/binding/org.openhab.binding.vera2/src/main/java/org/openhab/binding/vera2/internal/converter/VeraDeviceStateConverter.java

```diff
@@ -0,0 +1,177 @@
+                    case 1:
+                        return getDoorlockState(device.tripped.equals("1") ? "0" : "1");
+                    default:
+                        return getBinaryState(device.tripped);
+                }
+            case 5: // TODO HVAC
+                logger.warn("TODO: HVAC: {}, {}, {}, {}", device.id, device.name, device.category, device.categoryName);
+                break;
+            case 6: // TODO Camera
+                logger.warn("TODO: Camera: {}, {}, {}, {}", device.id, device.name, device.category,
+                        device.categoryName);
+                break;
+            case 7: // Door Lock
+                return getBinaryState(device.locked);
+            case 8: // Window Covering
+                return getPercentState(device.level);
+            case 9: // TODO Remote Control
+                logger.warn("TODO: Remote Control: {}, {}, {}, {}", device.id, device.name, device.category,
+                        device.categoryName);
+                break;
+            case 10: // TODO IR Transmitter
+                logger.warn("TODO: IR Transmitter: {}, {}, {}, {}", device.id, device.name, device.category,
+                        device.categoryName);
+                break;
+            case 11: // TODO Generic I/O
+                logger.warn("TODO: Generic I/O: {}, {}, {}, {}", device.id, device.name, device.category,
+                        device.categoryName);
+                break;
+            case 12: // Generic Sensor
+                return getMultilevelState(device.level);
+            case 13: // TODO Serial Port
+                logger.warn("TODO: Serial Port I/O: {}, {}, {}, {}", device.id, device.name, device.category,
+                        device.categoryName);
+                break;
+            case 14: // Scene Controller
+                return getBinaryState(device.status);
+            case 15: // TODO A/V
+                logger.warn("TODO: A/V: {}, {}, {}, {}", device.id, device.name, device.category, device.categoryName);
+                break;
+            case 16: // Humidity Sensor
+                return getMultilevelState(device.humidity);
+            case 17: // Temperature Sensor
+                return getMultilevelState(device.temperature);
+            case 18: // Light Sensor
+                return getMultilevelState(device.light);
+            case 19: // TODO Z-Wave Interface
+                logger.warn("TODO: Z-Wave Interface: {}, {}, {}, {}", device.id, device.name, device.category,
+                        device.categoryName);
+                break;
+            case 20: // TODO Insteon Interface
+                logger.warn("TODO: Insteon Interface: {}, {}, {}, {}", device.id, device.name, device.category,
+                        device.categoryName);
+                break;
+            case 21: // Power Meter
+                return getMultilevelState(device.level);
+            case 22: // TODO Alarm Panel

```

Comment:
Maybe you should make an ENUM or constant for these, it is a maintenance nightmare such big lists of meaningless number appearing in multiple parts of your code.